src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -27,7 +27,7 @@
      */
     public static final String APPENDIX_CHARACTERS = "abcdefghijklmnopqrstuvwxyz";
     private static final Logger LOGGER = LoggerFactory.getLogger(BibtexKeyGenerator.class);
-    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"-#~^:'`ʹ";
+    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"#~^:'`ʹ";
```

Comment:
```suggestion
    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"#%~'`ʹ";
```

I read the linked issues and especially https://tex.stackexchange.com/a/408548/9075. I adaopted the list accordingly (added `%` and removed `^`). Additionally, I left the quotes at the end.

Reply:
This update fixes https://github.com/JabRef/jabref/issues/6257.